Add multi-storage documentation (#255)
scalar-labs/scalardb: docs/multi-storage.md

# Multi-storage in Scalar DB

> Scalar DB supports transactions that span multiple storages/databases.

Reviewer: ```suggestion
Scalar DB transactions can span multiple storages/databases while preserving ACID property with a feature called `multi-storage`.
```

> This documentation explains this feature that's called `multi-storage`.

Reviewer: ```suggestion
This documentation explains the feature briefly.
```

## How Multi-storage works

Internally, the `multi-storage` implementation holds multiple storage instances and has mappings from a table name/a namespace name to a proper storage instance.

> When an operation executes, it chooses a proper storage instance from the specified table/namespace by using the table-storage/namespace-storage mappings and uses it.

Reviewer: ```suggestion
When an operation is executed, it chooses a proper storage instance from the specified table/namespace by using the table-storage/namespace-storage mappings and uses it.
```

## Configuration

> You can use `multi-storage` the same way as other storages/databases at the code level.

Reviewer: ```suggestion
You can use `multi-storage` in the same way as the other storages/databases at the code level as long as the configuration is properly set for `multi-storage`.
```

> On the other hand, the configuration is different.

Reviewer: ```suggestion
```

> An example of the configuration is as follows:

Reviewer: ```suggestion
An example of the configuration is shown as follows:
```
